strings/my_vsnprintf.c

```diff
@@ -749,23 +745,45 @@ size_t my_vsnprintf_ex(CHARSET_INFO *cs, char *to, size_t n,
       to= process_dbl_arg(to, end, width, d, *fmt);
       continue;
     }
-    else if (*fmt == 'd' || *fmt == 'i' || *fmt == 'u' || *fmt == 'x' || 
-             *fmt == 'X' || *fmt == 'p' || *fmt == 'o')
+    case 'd':
+    case 'i':
+    case 'u':
+    case 'x':
+    case 'X':
+    case 'o':
+    case 'p':
+    case 'M':
     {
       /* Integer parameter */
       longlong larg;
-
+      my_bool suffix_e= *fmt == 'M';
       if (have_longlong)
         larg = va_arg(ap,longlong);
-      else if (*fmt == 'd' || *fmt == 'i')
+      else if (*fmt == 'd' || *fmt == 'i' || suffix_e)
         larg = va_arg(ap, int);
       else
         larg= va_arg(ap, uint);
```

Comment:
Pessimistic backward compatibility – is there a world where `va_arg(ap, int)` and `va_arg(ap, uint)` differ?
```suggestion
      else
        larg = va_arg(ap, int);
```

Reply:
Positive integers > 2147483647

Reply:
Ah, right, may not between themselves, but yes in the context of assigning to (and later processing as) a `long long`.

It is in the original code that `%M` reads a `signed int`. `%uE` will of course read an `unsigned int`.